mounts.go

```diff
@@ -0,0 +1,173 @@
+package llblib
+
+import (
+	"fmt"
+
+	"github.com/moby/buildkit/client/llb"
+	"golang.org/x/exp/maps"
+)
+
+type MountPropagator interface {
+	Run(...llb.RunOption)
+	// Add will append a RunOption to the list of propagated options.
+	Add(...llb.RunOption)
+	ApplyRoot(...llb.StateOption)
+	ApplyMount(target string, opts ...llb.StateOption)
+	Root() llb.State
+	GetMount(target string) llb.State
+	AsRun() llb.RunOption
+	Copy() MountPropagator
+}
+
+func Persistent(root llb.State, opts ...llb.RunOption) MountPropagator {
+	pm := &persistentMounts{
+		opts:   opts,
+		root:   root,
+		states: map[string]llb.State{},
+	}
+	return pm
+}
+
+type persistentMounts struct {
+	opts   []llb.RunOption
+	root   llb.State
+	states map[string]llb.State
```

Comment:
Maybe this should be `map[string]*llb.State`, since `Copy` only makes a shallow copy of the states anyway (I assume that's intended?).

Reply:
Not sure about this, I dont think we really gain much by holding a pointer to an `llb.State`.  fwiw, the llb code always passes around an `llb.State` rather than `*llb.State`.

Reply:
I added a comment to Copy about the shallow copy also.